pkg/splunk/enterprise/configuration_test.go

```diff
@@ -595,134 +595,204 @@ func TestValidateSplunkSmartstoreSpec(t *testing.T) {
+	AppFramework.AppSources[0].VolName = ""
+	AppFramework.Defaults.VolName = ""
 
-	//validate that the S3PollInterval has been set to default value of 60 minutes
-	if AppFrameWorkDefaultS3PollInterval.S3PollInterval != 60 {
-		t.Errorf("The S3PollInterval did not get set to default value")
+	err = ValidateAppFrameworkSpec(&AppFramework, false)
+	if err == nil {
+		t.Errorf("If no default volume, App Source with missing volume info should return an error")
 	}
 
+	// If the AppSource doesn't have VolName, and if the defaults have it, shouldn't cause an error
+	AppFramework.Defaults.VolName = "msos_s2s3_vol"
+	err = ValidateAppFrameworkSpec(&AppFramework, false)
 	if err != nil {
-		t.Errorf("Expected no errors, the S3PollInterval would have set to default")
+		t.Errorf("If default volume, App Source with missing volume should not return an error, but got erros %v", err)
 	}
 
-	//Test6: FeatureEnabled is false so no validation errors expected
-	AppFrameWorkFeatureDisabled := enterprisev1.AppFrameworkSpec{
-		Type:           "S3",
-		S3Endpoint:     "http://test_s3_end_point",
-		S3SecretRef:    "appSecret",
-		S3PollInterval: 60,
+	// Volume referenced from an index must be a valid volume
+	AppFramework.AppSources[0].VolName = "UnknownVolume"
+
+	err = ValidateAppFrameworkSpec(&AppFramework, false)
+	if err == nil {
+		t.Errorf("Index with an invalid volume name should return error")
 	}
+	AppFramework.AppSources[0].VolName = "msos_s2s3_vol"
 
-	err = ValidateAppFrameworkSpec(&AppFrameWorkFeatureDisabled)
-	if err != nil {
-		t.Errorf("Expected no errors as FeatureEnabled is false but got error: %s", err)
+	// if the CR supports only local apps, and if the app source scope is not local, should return error
+	AppFramework.AppSources[0].Scope = "cluster"
+	err = ValidateAppFrameworkSpec(&AppFramework, true)
+	if err == nil {
+		t.Errorf("When called with App scope local, any app sources with the cluster scope should return an error")
 	}
 
-	//Test7: FeatureEnabled is true, validate no errors when all
-	//       parameters are provided
-	AppFrameWorkAllParamsValid := enterprisev1.AppFrameworkSpec{
-		Type:           "S3",
-		S3Endpoint:     "http://test_s3_end_point",
-		S3Bucket:       "test_bucket",
-		S3SecretRef:    "appSecret",
-		S3PollInterval: 60,
+	// If the app scope value other than "local" or "cluster" should return an error
+	AppFramework.AppSources[0].Scope = "unknown"
+	err = ValidateAppFrameworkSpec(&AppFramework, false)
+	if err == nil {
+		t.Errorf("Unsupported app scope should be cause error, but failed to detect")
 	}
 
-	err = ValidateAppFrameworkSpec(&AppFrameWorkAllParamsValid)
-	if err != nil {
-		t.Errorf("Expected no errors as all good params provided but got error: %s", err)
+	// If the CR supports only local apps, and default is configured with "cluster" scope, that should be detected
+	AppFramework.AppSources[0].Scope = "local"
+	AppFramework.AppSources[1].Scope = "local"
+	AppFramework.AppSources[2].Scope = "local"
```

Comment:
Super nit: Future we can have one more negative test case where anyone AppSources can have "cluster" scope defined and everything else is set to "local", for a CR that supports only local apps